packages/synapse-react-client/src/utils/hooks/useRefDimensions.ts

```diff
@@ -0,0 +1,46 @@
+import { RefObject, useEffect, useState } from 'react'
+
+/**
+ * How to use:
+   const divRef = createRef()
+   const dimensions = useRefDimensions(divRef)
+  ...
+  return (
+    <div style={{ height: '100vh', width: '100vw' }}>
+      <div
+        ref={divRef}
+      >
+        Dimensions: {dimensions.width}w {dimensions.height}h
+      </div>
+    </div>
+  )
+ */
+
+const useRefDimensions = (ref: RefObject<Element>) => {
+  const [dimensions, setDimensions] = useState({ width: 1, height: 2 })
+  useEffect(() => {
```

Comment:
Not 100% sure but this might be one of the rare cases where you want to use [useLayoutEffect](https://react.dev/reference/react/useLayoutEffect)

Reply:
I didn't see any improvement in the layout behavior when using useLayoutEffect(), so I'd rather just keep using useEffect()